### PR TITLE
Added suppressCompletionReranking configuration setting.

### LINKED ISSUE #37: Auto complete starting at incorrect entry
Please see history of closed issue on VSCode: https://github.com/microsoft/vscode/issues/98835

Here's a visual repro of the problem, with debug view enabled:

AutoComplete3

I'm trying to get `KeyHeld` to select so that I can press `tab` to auto complete. I have `"editor.suggestSelection":"first"` setup in my settings.

When I type `KeyH` it still selects `KeyStateHeld`. I have to type `KeyHeld` completely for it to highlight and enable tab completion.

VS Code version: Code 1.45.1 (5763d909d5f12fe19f215cbfdd29a91c0fa9208a, 2020-05-14T08:27:35.169Z)
OS version: Windows_NT x64 10.0.18362
clangd version: 10.0.0

<details>
<summary>System Info</summary>

|Item|Value|
|---|---|
|CPUs|Intel(R) Core(TM) i7-6800K CPU @ 3.40GHz (12 x 3398)|
|GPU Status|2d_canvas: enabled<br>flash_3d: enabled<br>flash_stage3d: enabled<br>flash_stage3d_baseline: enabled<br>gpu_compositing: enabled<br>multiple_raster_threads: enabled_on<br>oop_rasterization: disabled_off<br>protected_video_decode: unavailable_off<br>rasterization: enabled<br>skia_renderer: disabled_off_ok<br>video_decode: enabled<br>viz_display_compositor: enabled_on<br>viz_hit_test_surface_layer: disabled_off_ok<br>webgl: enabled<br>webgl2: enabled|
|Load (avg)|undefined|
|Memory (System)|31.91GB (18.32GB free)|
|Process Argv||
|Screen Reader|no|
|VM|0%|
</details><details><summary>Extensions (5)</summary>

Extension|Author (truncated)|Version
---|---|---
unicode-math-vscode|Gui|0.2.6
vscode-clangd|llv|0.1.5
cpptools|ms-|0.28.1
perforce|sle|3.2.0
vscode-jumpy|wma|0.3.1

</details>

### LINKED ISSUE #62: Allow returning isIncomplete from clangd instead of forcing it to true
Currently the vscode-clangd forces the isIncomplete flag to true, which makes autocompletion in large codebases (e.g. LLVM) fell sluggish. This is especially noticeable when developing on remote server.

Would you be interested in reviewing a PR adding a new setting "serverSideRanking" (set to true by default) which will disable the current isIncomplete override if set to false?

### PR DESCRIPTION
**Pull request by OlegOAndreev**
Added suppressCompletionReranking configuration setting to disable the forced clangd-based completion ranking added in https://github.com/OlegOAndreev/vscode-clangd/commit/5616b189db01094940a70db2e8314d9135ab9a10#diff-45327f86d4438556066de133327f4ca2

Motivation: for shorter completion lists (e.g. object methods after the dot/arrow) the forced round-trips to clangd add some perceptible UI lag (fixes #62, but may as well fix #37).

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -119,6 +119,11 @@
                     },
                     "description": "Extra clang flags used to parse files when no compilation database is found."
                 },
+                "clangd.suppressCompletionReranking": {
```
Comment (sam-mccall) [inline package.json:122]: I think the doulbe-negative feel (suppress + "reranking" implying undoing the old ranking) makes this a bit hard to understand. And the description only mentions one half of the tradeoff, so it's hard to make a choice.

What about `clangd.serverCompletionRanking` / "Always rank completion items on the server as you type. This produces more accurate results at the cost of higher latency than client-side filtering."

**Comment by sam-mccall** [comment]
I'm really sorry we let this fly by. I think it makes sense and we should merge it, just want to tweak the wording a little to ensure it's clear.

**Review by sam-mccall — CHANGES_REQUESTED** [review]

File: package.json (hunk 1 above)
Comment (OlegOAndreev) [reply package.json]: Thanks, I did not like the original name as well =) Fixed.

**Review by sam-mccall — APPROVED** [review]
Thank you!

**Pull request merged**